### ISSUE TITLE
UXD - Create button to reset selection mode and recenter around all boundaries

### ISSUE THREAD

**Issue by ryanfchase** [hackforla/311-data#1817]
### Overview
Users are getting lost in the map when selecting an NC, we need a way to quickly reset to the app's default state

### Action Items

- [x] Address the problem overview with possible design solution(s) in a comment
- [x] Complete Design Iterations section below
- [x] Document user interaction in Figma

### Resources

V1.2 Moderated Usability Study
- [October '23 slide indicating users expected a control for seeing entire map / attempting to unselect by zooming out](https://docs.google.com/presentation/d/1NBMnS8_znCnYNDNNQrN2Jh5GQZhZmX56rtHME-EewBY/edit#slide=id.g29f7a872fc9_0_73)
- [August '24 slide indicating users expected a control for seeing entire map](https://docs.google.com/presentation/d/1w1YpzC-sM7X1OOQ70f5Ey8h6Q0yOoTlC/edit#slide=id.p17)

---

### Design Iterations

**Please move ticket between `In Progress` and `In Review` to assist PM team**

<details><summary>Iteration 1</summary>
<p>

Link to notes: https://github.com/hackforla/311-data/issues/1817#issuecomment-2439233373

<img width="596" alt="Screenshot 2024-10-25 at 8 25 48 PM" src="https://github.com/user-attachments/assets/7a50a9de-6120-41fb-959a-7e649276fdc5">

</p>
</details> 

<details><summary>Iteration 2</summary>
<p>

Link to notes: https://github.com/hackforla/311-data/issues/1817#issuecomment-2613784710

<img width="1042" alt="Image" src="https://github.com/user-attachments/assets/2706d58b-e0cf-42d4-a401-0d0f4ea5a211" />

</p>
</details> 
---

### Hand Off Materials

Figma Section Name: [⭐⭐⭐ Handoff](https://www.figma.com/design/eF63O8yIGz2soQ4JJFfyoK/311_Data_UXUI?node-id=4081-9245)

<details><summary>Before Screenshot</summary>
<p>

<img width="299" alt="Image" src="https://github.com/user-attachments/assets/41a8ba3b-6615-4fc2-9f5a-b9f98cf6fd2b" />

</p>
</details> 

<details><summary>After Screenshot (Finalized)</summary>
<p>

<img width="311" alt="Image" src="https://github.com/user-attachments/assets/f6099502-1339-4e59-8a19-8a1e2f377e59" />

</p>
</details> 

### Designer Resources

<details><summary>Iteration Dropdown Copy/Paste</summary>
<p>

```
<details><summary>Iteration X</summary>
<p>

Link to notes: `REPLACE WITH COMMENT URL`

`REPLACE WITH SCREENSHOT UPLOAD`

</p>
</details> 
```

</p>
</details> 

<details><summary>Instructions for Engineering Hand Off</summary>
<p>

To Start Engineering Hand Off...
1. Ensure all Hand Off Materials are filled in
3. Add the "ready for dev lead" label
4. Leave a comment saying "This ticket is ready for engineering hand off." 

</p>
</details>

**Comment by ryanfchase** [comment]
Ready for design lead input. Please add:
- any further required information
- any possible blockers
- remove `Size: Missing` and add appropriate size

**Comment by pogwon** [comment]
After using different websites as examples and looking up best practices, most of the research for resetting filters seems to be within the context of e-commerce, but I think that's the closest we're going to get. I am designing a couple options and will have them ready by 10/16's meeting for a round of feedback.

**Comment by pogwon** [comment]
[Iteration 1](https://www.figma.com/design/eF63O8yIGz2soQ4JJFfyoK/311_Data_UXUI?node-id=14555-2130&t=xjhz7JV3sR6Z0Z6f-4)

After discussing with design team, two designs (top left and bottom left) seemed to make the most sense, so I will be going forward with those (along with seeing if any research favors one of them more) to finalize designs and make a decision.

<img width="596" alt="Screenshot 2024-10-25 at 8 25 48 PM" src="https://github.com/user-attachments/assets/54b12686-081f-482c-abdc-0ce0b5c15ca6">

**Comment by pogwon** [comment]
While working on second iterations, a lot of research shows that placing the clear/reset all buttons so close to another frequently used button (i.e. near the dropdown button for the neighborhood districts) could cause errors with users erasing accidentally erasing all their selections when they mean to only remove one. Are there designs that show what the dropdown would look like with multiple NC selections that we can use? @Joy-Truex @allisonjeon

**Comment by Joy-Truex** [comment]
@ryanfchase At the last meeting before the break we talked about the possibility that at this time we don't have the option to view multiple neighborhood councils at once. Can you confirm that we're assuming correctly about the current functionality going into the launch?

CC: @pogwon @allisonjeon

**Comment by ryanfchase** [comment]
> [@ryanfchase](https://github.com/ryanfchase) At the last meeting before the break we talked about the possibility that at this time we don't have the option to view multiple neighborhood councils at once. Can you confirm that we're assuming correctly about the current functionality going into the launch?
> 
> CC: [@pogwon](https://github.com/pogwon) [@allisonjeon](https://github.com/allisonjeon)

Yeah multiple NC selection is not slated for launch. Certainly there is buzz around such a feature (we used to have a mockup for this, there are old email communications marketing this idea).

**Comment by Joy-Truex** [comment]
@pogwon with Ryan's additional context added, my recommendation would be to add a "clear" button by the boundaries menu and/or a "reset all" button by the filters text. feel free to comment on the figma file/here with your thoughts!

**Comment by pogwon** [comment]
I'll work under constraint that only one NC at a time can be selected then. Including a 'reset all' by feels like it could be confused as applying to all filters (NC, date range, and request types), so I'll probably go with a button within the boundaries menu. I will update with designs, potential edge cases, and notes about interaction/behaviors by the end of this week. @Joy-Truex

**Comment by pogwon** [comment]
Decided to go with the 'reset' button within the Boundaries section. Updated with notes, interactions, and possible edge cases for handoff.

[Link to Figma file](https://www.figma.com/design/eF63O8yIGz2soQ4JJFfyoK/311_Data_UXUI?node-id=16104-5003&t=dTAaMqO8m4eSlYco-4)

Screenshot of design:

<img width="1042" alt="Image" src="https://github.com/user-attachments/assets/ff0a3423-6571-4143-b282-a28399aabe4e" />

**Comment by Joy-Truex** [comment]
Looks great @pogwon :)

This ticket is ready for engineering hand off!

**Comment by ryanfchase** [comment]
Closing this ticket as the dev ticket is being put together now. Thanks @pogwon !
